provider/src/test/kotlin/org/gradle/kotlin/dsl/support/ImplicitImportsTest.kt

```diff
@@ -0,0 +1,42 @@
+package org.gradle.kotlin.dsl.support
+
+import org.gradle.kotlin.dsl.fixtures.AbstractIntegrationTest
+import org.hamcrest.CoreMatchers.equalTo
+import org.junit.Assert.assertThat
+import org.junit.Test
+import java.io.File
+import java.util.jar.JarFile
+
+class ImplicitImportsTest : AbstractIntegrationTest() {
+
+    @Test
+    fun `implicit imports are fully qualified to allow use of the preferred type amongst those with same simple name in different Gradle API packages`() {
+
+        // given:
+        existing("settings.gradle").appendText("""
+            rootProject.name = "some-project"
+        """)
+        withBuildScript("""
+
+            plugins {
```

Comment:
:x: This test could be both simpler and more precise by asserting on the resolved type name and only that with something like:

```kotlin
println("*${Jar::class.qualifiedName}*")
```

And then checking the output for`*org.gradle.api.tasks.bundling.Jar*`.

Reply:
Done!